src/mscorlib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -1217,6 +1218,41 @@ public override int GetHashCode()
             return (this.Name.GetHashCode());
         }
 
+        internal static unsafe int GetIgnoreCaseHash(string source)
+        {
+            // Do not allocate on the stack if string is empty
+            if (source.Length == 0)
+            {
+                return source.GetHashCode();
+            }
+
+            Span<char> span = source.Length < 256 ? 
+                stackalloc char[source.Length] : 
+                ArrayPool<char>.Shared.Rent(source.Length);
```

Comment:
You are not returning the rented array back to the pool. You may want to just use ValueStringBuilder for this that takes care of it for you.

Reply:
argh right, I forgot that. I tend to not use VSB here, do you have strong opinion on that?

Reply:
Why you would not want to use it? If you have a good reason for it, feel free to not use it here.